modules/mono/glue/GodotSharp/GodotPlugins/PluginLoadContext.cs

```diff
@@ -21,6 +21,20 @@ public class PluginLoadContext : AssemblyLoadContext
             _resolver = new AssemblyDependencyResolver(pluginPath);
             _sharedAssemblies = sharedAssemblies;
             _mainLoadContext = mainLoadContext;
+
+            if (string.IsNullOrEmpty(AppContext.BaseDirectory))
+            {
+                // See https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/AppContext.AnyOS.cs#L18-L36
+                // but Assembly.Location is unavailable, because we load assemblies from memory.
+                string? baseDirectory = Path.GetDirectoryName(pluginPath);
+                if (baseDirectory != null)
+                {
+                    if (!Path.EndsInDirectorySeparator(baseDirectory))
+                        baseDirectory += Path.PathSeparator;
+                    // This SetData call effectively sets AppContext.BaseDirectory
+                    AppDomain.CurrentDomain.SetData("APP_CONTEXT_BASE_DIRECTORY", baseDirectory);
```

Comment:
Is this documented somewhere? If it's a private implementation, I'm afraid it may be changed later without warning (still be fine if there's no other way).

Reply:
As far as I can tell, it is not documented anywhere that the property can be set at all.

Well https://learn.microsoft.com/en-us/dotnet/api/system.appcontext.basedirectory?view=net-7.0#remarks points to the `BaseDirectory` property of AppDomain which is kinda implied to be able to be set as the `APPBASE` data (https://learn.microsoft.com/en-us/dotnet/api/system.appdomain.basedirectory?view=net-7.0#remarks) but that is (at least for .NET5+) not correct because it simply delegates to AppContext, which reads APP_CONTEXT_BASE_DIRECTORY. (and even on .NET framework, where setting APPBASE works, APP_CONTEXT_BASE_DIRECTORY overrides that.

The only other way to influence the BaseDirectory proeprty is to set the entry assembly (how? idk) and to set Assembly.Location (impossible without locking the assembly file)

I added a comment linking to the source of AppContext.BaseDirectory to explain the source of that constant.